### PR TITLE
Prevent entire bundle lookup in bundle validator

### PR DESCRIPTION
**Pull request by paulhandy**

### CODE REVIEW

File: src/main/java/com/iota/iri/controllers/TransactionViewModel.java [hunk 1]
```diff
@@ -160,11 +160,11 @@ public void delete(Tangle tangle) throws Exception {
 
     public List<Pair<Indexable, Persistable>> getMetadataSaveBatch() throws Exception {
         List<Pair<Indexable, Persistable>> hashesList = new ArrayList<>();
-        hashesList.add(new Pair<>(getAddressHash(), new Address(hash)));
-        hashesList.add(new Pair<>(getBundleHash(), new Bundle(hash)));
+        if(!getAddressHash().equals(Hash.NULL_HASH)) hashesList.add(new Pair<>(getAddressHash(), new Address(hash)));
+        if(getCurrentIndex() == 0) hashesList.add(new Pair<>(getBundleHash(), new Bundle(hash)));
```
Comment (alon-e) [inline src/main/java/com/iota/iri/controllers/TransactionViewModel.java]: 1. would make breaking changes to the API call `findTransactions` `bundles` - will only return tails, not all related hashes.
2. would require a DB migration (`--rescan`), not such an issue, just need to make sure it's clear.
--
we can postpone these changes (including not storing the `NULL_HASH`s) for another PR. it's not 100% related

File: src/main/java/com/iota/iri/controllers/TransactionViewModel.java [hunk 2]
```diff
@@ -160,11 +160,11 @@ public void delete(Tangle tangle) throws Exception {
 
     public List<Pair<Indexable, Persistable>> getMetadataSaveBatch() throws Exception {
         List<Pair<Indexable, Persistable>> hashesList = new ArrayList<>();
-        hashesList.add(new Pair<>(getAddressHash(), new Address(hash)));
-        hashesList.add(new Pair<>(getBundleHash(), new Bundle(hash)));
```
Comment (alon-e) [inline src/main/java/com/iota/iri/controllers/TransactionViewModel.java]: also, there is an implicit use of this `getMetadataSaveBatch` function and that is to fill the lazy fields in the transactionVM - this PR fails when storing non-tail transactions b/c of `bundle` bytes not being present:
```java.lang.NullPointerException: null
	at com.iota.iri.model.Transaction.metadata(Transaction.java:65) ~[classes/:na]
	at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.saveBatch(RocksDBPersistenceProvider.java:346) ~[classes/:na]
	at com.iota.iri.storage.Tangle.saveBatch(Tangle.java:53) ~[classes/:na]
	at com.iota.iri.controllers.TransactionViewModel.store(TransactionViewModel.java:207) ~[classes/:na]
	at com.iota.iri.network.Node.processReceivedData(Node.java:376) [classes/:na]
	at com.iota.iri.network.Node.processReceivedDataFromQueue(Node.java:358) [classes/:na]
	at com.iota.iri.network.Node.lambda$spawnProcessReceivedThread$8(Node.java:559) [classes/:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_151]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_151]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_151]```

Comment (alon-e) [reply src/main/java/com/iota/iri/controllers/TransactionViewModel.java]: tested reverting the changes to `TransactionViewModel.java` - works well.

**Comment by paulhandy** [comment]
Merged into v1.4.1.7

**Pull request closed without merging**